vim-patch:partial:cfa8f9a3f285

clason opened the pull request:
Fixes #18754

Update runtime files
https://github.com/vim/vim/commit/cfa8f9a3f285060152ebbdbf86fbc7aecf1dd756

omit:
* translations
* manual rewrite
* most doc changes (for now, @seandewar or @zeertzjq let me know if there's a specific change that is valid for Neovim, then I'll add it)

zeertzjq commented on runtime/doc/map.txt:
```diff
@@ -152,6 +151,22 @@ that mapping won't get expanded yet, Vim is waiting for another character.
 If you type a space, then "foo" will get inserted, plus the space.  If you
 type "a", then "bar" will get inserted.
 
+Trailing white space ~
+							*map-trailing-white*
+This unmap command does NOT work: >
+	:map @@ foo
+	:unmap @@ | print
+Because it tries to unmap "@@ ", including the white space before the command
+separator "|".  Other examples with trailing white space: >
+	unmap @@ 
+	unmap @@     " legacy comment
+An error will be issued, which is very hard to identify, because the ending
+whitespace character in `unmap @@ ` is not visible.
+
+A generic solution is to put the command separator "|" right after the mapped
+keys.  After that white space and a comment may follow: >
+	unmap @@|    " legacy scriptcomment
+
```
Is the removal of blank lines intentional?

clason replied:
No, for some reason they did not get copied from the Github diffview.

zeertzjq commented on runtime/doc/undo.txt:
```diff
@@ -108,12 +108,14 @@ change again.  But you can do something like this: >
+
 Setting the value of 'undolevels' also breaks undo.  Even when the new value
-is equal to the old value.
+is equal to the old value:: >
```
```suggestion
is equal to the old value: >
```

zeertzjq commented on runtime/syntax/vim.vim:
```diff
@@ -367,7 +367,7 @@ syn match	vimSetMod	contained	"&vim\=\|[!&?<]\|all&"
 " ===
 syn keyword	vimLet	let	unl[et]	skipwhite nextgroup=vimVar,vimFuncVar,vimLetHereDoc
-VimFoldh syn region vimLetHereDoc	matchgroup=vimLetHereDocStart start='=<<\s\+\%(trim\>\)\=\s*\z(\L\S*\)'	matchgroup=vimLetHereDocStop end='^\s*\z1\s*$'
+VimFoldh syn region vimLetHereDoc	matchgroup=vimLetHereDocStart start='=<<\s\+\%(trim\|eval\>\)\=\s*\z(\L\S*\)'	matchgroup=vimLetHereDocStop end='^\s*\z1\s*$'
```
Needs patch 8.2.4770

zeertzjq reviewed (commented):
Missing change.txt

clason committed:
vim-patch:partial: cfa8f9a3f285

Update runtime files
https://github.com/vim/vim/commit/cfa8f9a3f285060152ebbdbf86fbc7aecf1dd756

skip syntax/vim.vim (needs 8.2.4770)
skip further rewrite of manual

zeertzjq reviewed (approved):
Although syntax.txt change itself is applicable, there are too many previous patches that cannot be ported yet.

clason commented:
> Although syntax.txt change itself is applicable, there are too many previous patches that cannot be ported yet.

Yes, that whole help section is still missing.
